core-bundle/assets/controllers/scroll-offset-controller.js

```diff
@@ -0,0 +1,183 @@
+import { Controller } from '@hotwired/stimulus'
+
+export default class extends Controller {
+    static targets = ['scrollTo', 'autoFocus'];
+
+    static values = {
+        sessionKey: {
+            type: String,
+            default: 'contao_backend_offset'
+        },
+        addToAttribute: {
+            type: String,
+            default: 'data-add-to-scroll-offset'
+        },
+        behavior: {
+            type: String,
+            default: 'instant'
+        },
+        block: {
+            type: String,
+            default: 'center'
+        }
+    };
+
+    // Backwards compatibility: automatically register the Stimulus controller if the legacy methods are used
+    static afterLoad(identifier, application) {
+        const loadFallback = () => {
+            return new Promise((resolve, reject) => {
+                const controller = application.getControllerForElementAndIdentifier(document.documentElement, identifier);
+
+                if (controller) {
+                    resolve(controller);
+                    return;
+                }
+
+                const { controllerAttribute } = application.schema;
+                document.documentElement.setAttribute(controllerAttribute, `${document.documentElement.getAttribute(controllerAttribute) || ''} ${ identifier }`);
+
+                setTimeout(() => {
+                    const controller = application.getControllerForElementAndIdentifier(document.documentElement, identifier);
+                    controller && resolve(controller) || reject(controller);
+                }, 100);
+            });
+        }
+
+        if (window.Backend && !window.Backend.initScrollOffset) {
+            window.Backend.initScrollOffset = () => {
+                console.warn('Backend.initScrollOffset() is deprecated. Please use the Stimulus controller instead.');
+                loadFallback();
+            }
+        }
+
+        if (window.Backend && !window.Backend.getScrollOffset) {
+            window.Backend.getScrollOffset = () => {
+                console.warn('Backend.getScrollOffset() is deprecated. Please use the Stimulus controller instead.');
+                loadFallback().then((controller) => controller.discard());
+            }
+        }
+    }
+
+    initialize () {
+        this.store = this.store.bind(this);
+    }
+
+    connect () {
+        if (this.offset) {
+            window.scrollTo({
+                top: this.offset + this.additionalOffset,
+                behavior: this.behaviorValue,
+                block: this.blockValue
+            });
+
+            this.offset = null;
+        }
+
+        this.buttons = document.querySelectorAll('.tl_submit_container button[name]:not([name="save"])');
+
+        this.buttons.forEach((button) => {
+            button.addEventListener('click', this.store, { passive: true });
```

Comment:
```suggestion
            button.addEventListener('click', this.discard, { passive: true });
```

Should this not be `this.discard`, so the offset is removed? (Also in line 86.)